cpp/include/cudf/wrappers/bool.hpp

```diff
@@ -45,10 +45,15 @@ struct bool8 {
   bool8& operator=(bool8&&) = default;
   bool8& operator=(const bool8&) = default;
 
+  CUDA_HOST_DEVICE_CALLABLE constexpr bool8(bool v)
```

Comment:
This is an implicit constructor for any type that can be implicitly cast to boolean. The other constructor has to be made explicit again, otherwise the compiler warns about using `is_arithmetic`.

Reply:
Ah, okay. In that case... is there a need for the explicit constructor? Would it be sufficient to just have the implicit constructor from bool? That would allow any type that is convertible to bool.

Reply:
Though it's worth noting I can remove the `enable_if_t<std::arithmetic>` and everything still works, so I'm not sure exactly what it's guarding against when it coexists with the implicit constructor.

Reply:
Looks like the only difference caused by removing the explicit constructor is one `-Werror=int-in-bool-context` warning in the `operator *` implementation, but I don't think it should impact `bool8` consumers since we're overloading the multiplication operator.